### PR TITLE
Fix bug 1215670 - Correct multiword tag form processing

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

**Comment by brhom300** [comment]
http://www.microsoft.com/october2015event/ar-us

File: kuma/wiki/tests/test_forms.py [hunk 1]
```diff
@@ -105,6 +105,17 @@ def test_form_rejects_empty_slugs_with_parent(self):
         rev_form = RevisionForm(data, parent_slug='User:groovecoder')
         ok_(not rev_form.is_valid())
 
+    def test_multiword_tags(self):
+        rev = revision(save=True)
+        data = {
+            'content': 'Content',
+            'toc_depth': 1,
+            'tags': '"MDN Meta"',  # Note the lower-case "S".
```
Comment (jezdez) [inline kuma/wiki/tests/test_forms.py]: Code comment isn't accurate

**Comment by jezdez** [comment]
Hm, will this work with the current set of tags?

File: kuma/wiki/tests/test_forms.py (hunk 1 above)
Comment (robhudson) [inline kuma/wiki/tests/test_forms.py]: Oh, woops :)

**Comment by jezdez** [comment]
It's odd, that we have to do this parsing on our own, for some reason. Is there some features in django-taggit that we can re-use?

**Comment by robhudson** [comment]
> Hm, will this work with the current set of tags?

I'm not exactly sure what you mean? I tested a few commonly used tags like "MDN Meta". But once they page is saved and they are split apart b/c of this bug, this code will not fix them. It would require the writers to clean this up, which was mentioned in bugzilla. Which is why I would like to merge this soon.

> It's odd, that we have to do this parsing on our own, for some reason. Is there some features in django-taggit that we can re-use?

I haven't seen any code to help with this, and this is code that could be later removed with a move to Postgresql which is a code comment.

django-taggit has a setting, `TAGGIT_CASE_INSENSITIVE` which defaults to `False` but we don't want case insensitive tags. It basically switches to an `__iexact` instead of an `__in` query. See https://github.com/alex/django-taggit/blob/develop/taggit/managers.py#L163

This is basically working around the bug in django/mysql that `__in` and `__exact` are case insensitive to begin with.

**Comment by jezdez** [comment]
Ok, thanks for the explanation.

**Comment by jezdez** [comment]
r+

**Pull request merged**